### PR TITLE
existing-payment-options: tweaks to Direct Debit cloning

### PR DESCRIPTION
**Pull request by twrichards**
There was a long delay before I merged https://github.com/guardian/support-frontend/pull/1772 and another PR snuck in with a pattern match on `PaymentMethod`, which then became non-exhaustive when I finally merged https://github.com/guardian/support-frontend/pull/1772.

_NOTE, this only affects re-use of Direct Debit payment methods for contributions (for signed in users with existing products) but this was **already disabled** because of the support switches. Once this PR is merged and tested, hopefully we can turn this on._

Unfortunately it's not possible to make SBT error on **only** non-exhaustive pattern matches (it's either make all warnings into errors or none - which is a real shame) - this would've caught this in the merge build (i.e. failed and therefore wouldn't be deployable)
https://stackoverflow.com/questions/28327383/make-compile-fail-on-non-exhaustive-match-in-sbt  
... so **instead** I've made `MatchError` map to `RetryNone` (to prevent something that will never succeed from being retried - especially important if that thing has side-effects, like the mutations in GoCardless added in https://github.com/guardian/support-frontend/pull/1772).

Also **renamed** `GetPaymentMethod` lambda to `PreparePaymentMethodForReuse` (now that it does some mutation in GoCardless the name was misleading).

Also corrected the fields passed to the email to ensure parity with a brand new direct debit.

### CODE REVIEW

**Comment by twrichards** [comment]
Unfortunately it's not possible to make SBT error on **only** non-exhaustive pattern matches (it's either make all warnings into errors or none - which is a real shame) - this would've caught this in the merge build (i.e. failed and therefore wouldn't be deployable)
https://stackoverflow.com/questions/28327383/make-compile-fail-on-non-exhaustive-match-in-sbt

**Comment by johnduffell** [comment]
yes the error on warning is really annoying, it basically works really well on a new project, but as soon as you need to update play or scala or any other major library, you have to turn it off until you can fix all the deprecation warnings.

File: support-workers/src/main/scala/com/gu/support/workers/lambdas/PaymentMethodExtensions.scala [hunk 1]
```diff
@@ -1,13 +1,19 @@
 package com.gu.support.workers.lambdas
 
-import com.gu.support.workers.{CreditCardReferenceTransaction, DirectDebitPaymentMethod, PayPalReferenceTransaction, PaymentMethod}
+import com.gu.support.workers.{
```
Comment (johnduffell) [inline support-workers/src/main/scala/com/gu/support/workers/lambdas/PaymentMethodExtensions.scala]: interesting way to write imports, is that normal?

Comment (twrichards) [reply support-workers/src/main/scala/com/gu/support/workers/lambdas/PaymentMethodExtensions.scala]: just it was complaining about the line length, but wasn't sure whether to a an `_` import. Thoughts?

**Comment by twrichards** [comment]
TESTED in CODE ✅

File: support-workers/src/main/scala/com/gu/support/workers/lambdas/PaymentMethodExtensions.scala (hunk 1 above)
Comment (johnduffell) [reply support-workers/src/main/scala/com/gu/support/workers/lambdas/PaymentMethodExtensions.scala]: fine by me! just looked funny.  I'm not a big fan of _ imports anyway but intellij used to revert to them pretty quickly

**Pull request merged**